### PR TITLE
[bug] Fix parsing of latest OLSR jsoninfo format

### PR DESCRIPTION
**Pull request by wnagele**
OLSR jsoninfo plugin has changed the JSON output format in some of the "newer" releases. This adds support for parsing that - while remaining backward compatible.

### CODE REVIEW

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Hi @wnagele,

thank you for contributing!

Can you please provide a sample of the new format, so that we can update the automated tests?

**Comment by wnagele** [comment]
I've ported the `links2` OLSR data into the new format and changed to run every testcase that has been using it to run for both the old and new format. This should give decent test coverage. Hope this is what you were after.

**Comment by wnagele** [comment]
ping @nemesisdesign

**Review by nemesisdesign — COMMENTED** [review]
Thanks, we will test / review and do any improvements if needed asap.

File: tests/test_olsr.py [hunk 1]
```diff
@@ -18,8 +22,16 @@
 ).read()
 
 
+def parameterized_test_name_func(testcase_func, param_num, param):
+    format = 'newformat' if 'version' in param[0][0] else 'oldformat'
+    return f'{testcase_func.__name__}_{param_num}_{format}'
```
Comment (pandafy) [inline tests/test_olsr.py:27]: I have added this to make the name of test case readable in verbose mode. Without this, the whole content of `param` was being appended to the test case name in verbose mode.

**Review by nemesisdesign — APPROVED** [review]

**Comment by nemesisdesign** [comment]
Thanks @wnagele :+1:

**Pull request merged**